src/sdkBase.ts

```diff
@@ -295,8 +299,9 @@ export abstract class SdkBase {
     if (this.hasOptedOut()) this._callbackManager.runCallbacks(EventType.OptoutReceived, {});
   }
 
-  private isLoggedIn() {
-    return this._identity && !hasExpired(this._identity.refresh_expires);
+  private isIdentityValid() {
+    const identity = this._identity ?? this.getIdentityNoInit();
+    return identity && !hasExpired(identity.refresh_expires);
```

Comment:
Renamed function since its private and this makes more sense. Also added the `getIdentityNoInit` because we can have an identity without `initComplete` from the recent SDK changes